java/src/com/ibm/streamsx/topology/generator/spl/SPLGenerator.java

```diff
@@ -166,18 +581,40 @@ void generateComposite(JsonObject graphConfig, JsonObject graph,
         compBuilder.append("composite ");
 
         compBuilder.append(kind);
-        if (kind.startsWith("__parallel_")) {
-            String iput = jstring(graph, "inputName");
-            String oput = jstring(graph, "outputName");
-
-            iput = getSPLCompatibleName(iput);
-            compBuilder.append("(input " + iput);
+        if (kind.startsWith("Composite")) {
```

Comment:
Hmm, should we use a name here that is not a valid operator kind name? Ie. a customer could in theory have a valid primitive or SPL composite called 'Composite'.

Maybe use '$Composite' and should it be in a static field as a constant?

Reply:
Hmm, we need an identifier here, but we also need to ensure we don't clash with an user provided identifier.

Reply:
Might - delay this and fix in a subsequent PR.

Reply:
We don't need to keep the info in the kind name. I created a boolean field called `generated`, which is `true` if the composite was generated during SPL generation.